src/modules/app/app.ts

```diff
@@ -205,6 +205,17 @@ export class App {
                 this.searchParams.clearUrl();
                 this.globalFiltres.clearFilters();
             }
+            case 'buy': {
+                const paymentForm = new PaymentForm(MAIN_CONTAINER, this.onButtonClick);
+                paymentForm.drawForm();
+                break;
+            }
+            case 'pay': {
+                console.log('pay');
+                document.querySelector('.modal-window')!.innerHTML = `<p class="message">The order accepted!</p>`;
+                setTimeout(() => window.location.href = window.location.origin, 3000);
```

Comment:
Careful here. User might navigate (change location) within this 3 seconds interval. Then callback execution will be looking weird.

Reply:
Fixed